Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/SyncProgramsJobService.java

```diff
@@ -0,0 +1,164 @@
+package org.dolphinemu.dolphinemu.services;
+
+import android.app.job.JobParameters;
+import android.app.job.JobService;
+import android.content.ContentUris;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.PersistableBundle;
+import android.support.media.tv.Channel;
+import android.support.media.tv.PreviewProgram;
+import android.support.media.tv.TvContractCompat;
+import android.util.Log;
+
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.model.GameFile;
+import org.dolphinemu.dolphinemu.ui.platform.Platform;
+import org.dolphinemu.dolphinemu.utils.AppLinkHelper;
+import org.dolphinemu.dolphinemu.utils.TvUtil;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SyncProgramsJobService extends JobService
+{
+	private static final String TAG = "SyncProgramsJobService";
+
+	private SyncProgramsTask mSyncProgramsTask;
+
+	@Override
+	public boolean onStartJob(final JobParameters jobParameters)
+	{
+		Log.d(TAG, "onStartJob(): " + jobParameters);
+		final long channelId = getChannelId(jobParameters);
+		if (channelId == -1L)
+		{
+			Log.d(TAG, "Failed to find channel");
+			return false;
+		}
+
+		mSyncProgramsTask =
+			new SyncProgramsTask(getApplicationContext())
+			{
+				@Override
+				protected void onPostExecute(Boolean finished)
+				{
+					super.onPostExecute(finished);
+					mSyncProgramsTask = null;
+					jobFinished(jobParameters, !finished);
+				}
+			};
+		mSyncProgramsTask.execute(channelId);
+		return true;
+	}
+
+	@Override
+	public boolean onStopJob(JobParameters jobParameters)
+	{
+		if (mSyncProgramsTask != null)
+		{
+			mSyncProgramsTask.cancel(true);
+		}
+		return true;
+	}
+
+	private long getChannelId(JobParameters jobParameters)
+	{
+		PersistableBundle extras = jobParameters.getExtras();
+		return extras.getLong(TvContractCompat.EXTRA_CHANNEL_ID, -1L);
+	}
+
+	private class SyncProgramsTask extends AsyncTask<Long, Void, Boolean>
+	{
+		private Context context;
+		private List<GameFile> updatePrograms;
+
+		private SyncProgramsTask(Context context)
+		{
+			this.context = context.getApplicationContext();
+			updatePrograms = new ArrayList<>();
+			GameFileCacheService.startLoad(context);
```

Comment:
Why is this needed? and if it is needed, it doesn't seem we are syncing them together, meaning we are not waiting until the GameFileCacheService to finish before we start this one.

Reply:
You're right, this isn't needed. Updating the programs only occurs if they open the app or when they add a new game directory. At both points the games cache should already be initialized. Fixed.